packages/stats/src/collector.ts

```diff
@@ -0,0 +1,109 @@
+import * as Parser from '@tracespace/parser'
+import {DrillUsage, DrillStats} from './types'
+
+/**
+ * Method to collect drill stats of parsed drill files.
+ *
+ * @example
+ * ```ts
+ * import {createParser} from '@tracespace/parser'
+ * import {collectDrillStats} from '@tracespace/stats'
+ *
+ * const parser = createParser()
+ *
+ * parser.feed(...)
+ *
+ * const tree = parser.results()
+ * const stats = collectDrillStats([tree])
+ * ```
+ *
+ * @category Stats
+ */
+export function collectDrillStats(trees: Parser.Root[]): DrillStats {
+  let toolsMap = new Map<string, Parser.ToolShape>()
+  let drillsPerTool = new Map<string, number>()
+  let routesPerTool = new Map<string, number>()
```

Comment:
These can all be `const`, because their reference doesn't change. And if we're just keying by a string, I think an object literal is just a little more my speed.

```suggestion
  const toolsMap: Partial<Record<string, Parser.ToolShape>> = {}
  const drillsPerTool: Partial<Record<string, number>> = {}
  const routesPerTool: Partial<Record<string, number>> = {}
```

I suggest `Partial<Record<...>>` as the TypeScript type because it enforces that every time you get a value, the typechecker will know it may be `undefined`